src/Data/List/Membership/Setoid/Properties.agda

```diff
@@ -363,6 +388,33 @@ module _ (S : Setoid c ℓ) {P : Pred (Carrier S) p}
   ...   | here  v≈x   = here v≈x , resp (sym v≈x) (invert [Px])
   ...   | there v∈fxs = Product.map there id (∈-filter⁻ v∈fxs)
 
+------------------------------------------------------------------------
+-- map∘filter
+
+module _
+  (S₁ : Setoid c₁ ℓ₁) (S₂ : Setoid c₂ ℓ₂)
+  {P : Pred (S₁ .Carrier) p} (P? : Decidable P) (resp : P Respects (S₁ .Setoid._≈_))
+  {f xs y} where
+
+  open Setoid     S₁ renaming (_≈_ to _≈₁_)
```

Comment:
Ugh! Is it perhaps cleaner to `open Setoid S₁` first?
```suggestion
  (open Setoid S₁ renaming (Carrier to A; _≈_ to _≈₁_))
  {P : Pred A p} (P? : Decidable P) (resp : P Respects _≈₁_)
  {f xs y} where

```

Reply:
Unfortunately, introducing `` in the module telescope (instead of a statement in the body of the module) leads to this cryptic error?!
```agda
module _
  (S₁ : Setoid c₁ ℓ₁) (S₂ : Setoid c₂ ℓ₂)
  (open Setoid S₁ renaming (Carrier to A; _≈_ to _≈₁_))
  {P : Pred _ p} (P? : Decidable P) (resp : P Respects _)
  {f xs y} where

  open Setoid     S₂ renaming (_≈_ to _≈₂_; sym to sym₂)
...
  ∈-map∘filter⁺ : f Preserves _≈₁_ ⟶ _≈₂_ →
```
```
Carrier S₁ !=< Data.List.Membership.Setoid.Properties.GeneralizeTel
when checking that the expression _≈₁_ has type
Rel (Carrier S₁) _ℓ₁_1616
```

The best I could come up with is using an extra inner module:
```agda
module _ (S₁ : Setoid c₁ ℓ₁) (S₂ : Setoid c₂ ℓ₂) where
  open Setoid S₁ renaming (Carrier to A; _≈_ to _≈₁_)
  open Setoid S₂ renaming (Carrier to B; _≈_ to _≈₂_; sym to sym₂)
  module _ {P : Pred A p} (P? : Decidable P) (resp : P Respects _≈₁_) {f xs y} where
    open Membership S₁ renaming (_∈_ to _∈₁_)
    open Membership S₂ renaming (_∈_ to _∈₂_)

    ∈-map∘filter⁻ : y ∈₂ map f (filter P? xs) →
                    ∃[ x ] x ∈₁ xs × y ≈₂ f x × P x
    ∈-map∘filter⁻ h =
      let x , x∈ , y≈ = ∈-map⁻ S₁ S₂ h
          y∈ , Py     = ∈-filter⁻ S₁ P? resp x∈
      in x , y∈ , y≈ , Py

    ∈-map∘filter⁺ : f Preserves _≈₁_ ⟶ _≈₂_ →
                    ∃[ x ] x ∈₁ xs × y ≈₂ f x × P x →
                    y ∈₂ map f (filter P? xs)
    ∈-map∘filter⁺ pres (x , x∈ , y≈ , Px)
      = ∈-resp-≈ S₂ (sym₂ y≈) 
      $ ∈-map⁺ S₁ S₂ pres (∈-filter⁺ S₁ P? resp x∈ Px)
```

Or, alternatively, letting Agda infer ` _≈₁_`:
```agda
module _
  (S₁ : Setoid c₁ ℓ₁) (S₂ : Setoid c₂ ℓ₂)
  (open Setoid S₁ renaming (Carrier to A))
  {P : Pred A p} (P? : Decidable P) (resp : P Respects _)
  {f xs y} where

  open Setoid     S₁ renaming (_≈_ to _≈₁_)
  open Setoid     S₂ renaming (_≈_ to _≈₂_; sym to sym₂)
  open Membership S₁ renaming (_∈_ to _∈₁_)
  open Membership S₂ renaming (_∈_ to _∈₂_)

  ∈-map∘filter⁻ : y ∈₂ map f (filter P? xs) →
                  ∃[ x ] x ∈₁ xs × y ≈₂ f x × P x
  ∈-map∘filter⁻ h =
    let x , x∈ , y≈ = ∈-map⁻ S₁ S₂ h
        y∈ , Py     = ∈-filter⁻ S₁ P? resp x∈
    in x , y∈ , y≈ , Py

  ∈-map∘filter⁺ : f Preserves _≈₁_ ⟶ _≈₂_ →
                  ∃[ x ] x ∈₁ xs × y ≈₂ f x × P x →
                  y ∈₂ map f (filter P? xs)
  ∈-map∘filter⁺ pres (x , x∈ , y≈ , Px)
    = ∈-resp-≈ S₂ (sym₂ y≈)
    $ ∈-map⁺ S₁ S₂ pres (∈-filter⁺ S₁ P? resp x∈ Px)
```
...and to avoid the telescope import altogether:
```agda
module _
  (S₁ : Setoid c₁ ℓ₁) (S₂ : Setoid c₂ ℓ₂)
  {P : Pred _ p} (P? : Decidable P) (resp : P Respects _)
  {f xs y} where
```

Shall I go with the last one?

Reply:
Ugh! That's really unfortunate... (caused by a clash between `variable` generalisation at the top of the module, and locally via the `open`ing of `Setoid`?)

... I'd be happy with the last approach, but again, I'd go with the view of the second reviewer, so leaving this as 'unresolved' for the time being...